Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java

```diff
@@ -0,0 +1,428 @@
+import edu.asu.conceptpower.servlet.db4o.DBNames;
+import edu.asu.conceptpower.servlet.db4o.IConceptDBManager;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryDoesNotExistException;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryEntryExistsException;
+import edu.asu.conceptpower.servlet.exceptions.DictionaryModifyException;
+import edu.asu.conceptpower.servlet.exceptions.LuceneException;
+import edu.asu.conceptpower.servlet.lucene.ILuceneUtility;
+import edu.asu.conceptpower.servlet.rest.LuceneFieldNames;
+import edu.asu.conceptpower.servlet.rest.SearchFieldNames;
+import edu.asu.conceptpower.servlet.wordnet.Constants;
+import edu.asu.conceptpower.servlet.wordnet.WordNetManager;
+
+/**
+ * This class handles concepts in general. It uses a DB4O client that contains
+ * all added concepts and concept wrappers, and a WordNet client that queries
+ * WordNet.
+ * 
+ * @author Julia Damerow
+ * 
+ */
+@Service
+public class ConceptManager implements IConceptManager {
+
+    @Autowired
+    private IConceptDBManager client;
+
+    @Autowired
+    private ILuceneUtility luceneUtility;
+
+    @Autowired
+    private WordNetManager wordnetManager;
+
+    protected final String CONCEPT_PREFIX = "CON";
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.conceptpower.core.IConceptManager#getConceptEntry(java.lang.
+     * String)
+     */
+    @Override
+    public ConceptEntry getConceptEntry(String id) throws LuceneException {
+        ConceptEntry entry = client.getEntry(id);
+        if (entry != null) {
+            fillConceptEntry(entry);
+            return entry;
+        }
+
+        entry = wordnetManager.getConcept(id);
+        if (entry != null) {
+            // client.store(entry, DBNames.WORDNET_CACHE);
+            return entry;
+        }
+
+        return null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.conceptpower.core.IConceptManager#getWordnetConceptEntry(java.
+     * lang.String)
+     */
+    @Override
+    public ConceptEntry getWordnetConceptEntry(String wordnetId) throws LuceneException {
+        ConceptEntry entry = getConceptEntry(wordnetId);
+        return entry;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.conceptpower.core.IConceptManager#getConceptListEntriesForWord(
+     * java.lang.String, java.lang.String)
+     */
+    @Override
+    public ConceptEntry[] getConceptListEntriesForWord(String word, String pos, String conceptType)
+            throws LuceneException, IllegalAccessException {
+        if (pos == null)
+            return null;
+        word = word.replace(" ", "");
+        Map<String,String> fieldMap = new HashMap<String,String>();
+        fieldMap.put(SearchFieldNames.WORD, word);
+        fieldMap.put(SearchFieldNames.POS, pos);
+        fieldMap.put(LuceneFieldNames.CONCEPTTYPE, conceptType);
+        return luceneUtility.queryIndex(fieldMap, null);
```

Comment:
why do you use lucenefieldnames here? and why did you add a concepttype? If you want to be able to search all wordnet concepts (from your calls it looks liek it), you can use the conceptlist property.